lib/Sema/TypeCheckPropertyWrapper.cpp

```diff
@@ -650,16 +678,25 @@ Expr *swift::buildPropertyWrapperInitialValueCall(
 
     SourceLoc startLoc = wrapperAttrs[i]->getTypeLoc().getSourceRange().Start;
 
+    auto init = var->getAttachedPropertyWrapperTypeInfo(i).wrappedValueInit;
+    if (init && resultTy) {
+      auto paramsList = init->getParameters();
+      if (paramsList->size() > 0 && paramsList->get(0)->isAutoClosure()) {
+        auto fnTy = FunctionType::get({}, resultTy);
+        initializer = new (ctx) AutoClosureExpr(initializer, fnTy,
```

Comment:
I'm surprised that you need to create the `AutoClosureExpr` at all here, because usually the type checker would introduce it itself. Do we need this particular code?

Reply:
Indeed, it appears that we don't need this code. I'll remove it. (I didn't know it would be taken care of by the type checker.)

Reply:
Great!